core/azure-core/src/main/java/com/azure/core/polling/Poller.java

```diff
@@ -0,0 +1,240 @@
+     */
+    public void cancelOperation() throws UnsupportedOperationException {
+        if (cancelOperation == null) {
+            throw new UnsupportedOperationException("Cancel operation is not supported on this service/resource.");
+        }
+
+        // We can not cancel an operation if it was never started
+        // or it is in its terminal state.
+        // Check cancelInitiated: to protect against multiple time call to cancel operation
+        if (cancelInitiated || (pollResponse != null && pollResponse.getStatus() != PollResponse.OperationStatus.IN_PROGRESS)) {
+            return;
+        }
+        //Time to call cancel
+        cancelOperation.accept(this);
+        cancelInitiated = true;
+    }
+
+    /**
+     * Enable client to subscribe and receive all the responses.
+     * @return Return poll response as Flux
+     */
+    public Flux<PollResponse<T>> getObserver() {
+         createFlux();
+         return fluxHandle;
+    }
+
+    /**
+     * Calls poll operation once in sync.
+     * @return  Mono of poll response
+     */
+    public Mono<PollResponse<T>> poll() {
+        updatePollOperationSync();
+        return Mono.just(pollResponse);
+    }
+
+    /**
+     * This will block till poll operation is complete
+     * @return returns poll response
+     */
+    public PollResponse<T> block() {
+        return getObserver().blockLast();
+    }
+
+    /*
+     * Calls poll operation function and update pollResponse.
+     */
+    private  void updatePollOperationSync() {
+        pollResponse = pollOperation.apply(pollResponse);
+    }
+
+    /*
+     * Get whether or not this PollStrategy's long running operation is done.
+     * @return Whether or not this PollStrategy's long running operation is done.
+     */
+    private Mono<PollResponse<T>> sendPollRequestWithDelay() {
+        return Mono.defer(() -> delayAsync().then(Mono.defer(() -> {
+            if (!isTerminalState()) {
+                updatePollOperationSync();
+            } else if (!isTerminalState()) {
+                return Mono.empty();
+            }
+            return Mono.just(pollResponse);
+        })));
+    }
+
+    /*
+     * Helper function
+     */
+    private boolean isTerminalState() {
+        return pollResponse != null && pollResponse.isDone();
+
+    }
+
+    /*
+     * If this PollerOptions has a pollIntervalInMillis value, return an Mono that is delayed by the
+     * number of seconds that are in the pollIntervalInMillis value. If this PollerOptions doesn't have
+     * a pollIntervalInMillis value, then return an Single with no delay.
+     *
+     * @return A Mono with delay if this PollerOptions has a pollIntervalInMillis value.
+     */
+    private Mono<Void> delayAsync() {
+        Mono<Void> result = Mono.empty();
+        if (this.pollerOptions.pollInterval().toNanos() > 0) {
+            result = result.delaySubscription(this.pollerOptions.pollInterval());
```

Comment:
What's the reason of choosing delaySubscription() over delay()? Maybe put a comment here since many readers could have the same question.